cmd/cli/kubectl-kyverno/processor/result.go

```diff
@@ -84,37 +84,56 @@ func (rc *ResultCounts) addGenerateResponse(response engineapi.EngineResponse) {
 }
 
 func (rc *ResultCounts) addMutateResponse(response engineapi.EngineResponse) bool {
+	printMutatedRes := false
 	genericPolicy := response.Policy()
-	if genericPolicy.AsKyvernoPolicy() == nil {
-		return false
-	}
-	policy := genericPolicy.AsKyvernoPolicy()
-	var policyHasMutate bool
-	for _, rule := range autogen.Default.ComputeRules(policy, "") {
-		if rule.HasMutate() {
-			policyHasMutate = true
+
+	// Handle Kyverno mutate policies
+	if kyvernoPolicy := genericPolicy.AsKyvernoPolicy(); kyvernoPolicy != nil {
+		// Check if it has at least one mutate rule
+		hasMutate := false
+		for _, rule := range autogen.Default.ComputeRules(kyvernoPolicy, "") {
+			if rule.HasMutate() {
+				hasMutate = true
+				break
+			}
 		}
-	}
-	if !policyHasMutate {
-		return false
-	}
-	printMutatedRes := false
-	for _, policyRule := range autogen.Default.ComputeRules(policy, "") {
-		for _, mutateResponseRule := range response.PolicyResponse.Rules {
-			if policyRule.Name == mutateResponseRule.Name() {
-				if mutateResponseRule.Status() == engineapi.RuleStatusPass {
-					rc.Pass++
-					printMutatedRes = true
-				} else if mutateResponseRule.Status() == engineapi.RuleStatusSkip {
-					rc.Skip++
-				} else if mutateResponseRule.Status() == engineapi.RuleStatusError {
-					rc.Error++
-				} else {
-					rc.Fail++
+		if !hasMutate {
+			return false
+		}
+
+		for _, policyRule := range autogen.Default.ComputeRules(kyvernoPolicy, "") {
+			for _, responseRule := range response.PolicyResponse.Rules {
+				if policyRule.Name == responseRule.Name() {
+					switch responseRule.Status() {
+					case engineapi.RuleStatusPass:
+						rc.Pass++
+						printMutatedRes = true
+					case engineapi.RuleStatusFail:
+						rc.Fail++
+					case engineapi.RuleStatusSkip:
+						rc.Skip++
+					case engineapi.RuleStatusError:
+						rc.Error++
+					}
 				}
-				continue
 			}
 		}
+		return printMutatedRes
+	}
+
+	// Handle native Kubernetes MAPs (no AsKyvernoPolicy)
+	for _, rule := range response.PolicyResponse.Rules {
+		switch rule.Status() {
+		case engineapi.RuleStatusPass:
+			rc.Pass++
+			printMutatedRes = true
+		case engineapi.RuleStatusFail:
+			rc.Fail++
+		case engineapi.RuleStatusSkip:
+			rc.Skip++
+		case engineapi.RuleStatusError:
+			rc.Error++
+		}
 	}
```

Comment:
You need to check that the policy is of type MAPs as you did here:
https://github.com/kyverno/kyverno/blob/0ec1222ef0cbb94e8e738bd32ef4916d1191938b/cmd/cli/kubectl-kyverno/processor/result.go#L91